### ISSUE TITLE
Support optimisers storing interim results

### ISSUE THREAD

**Issue by eileen-kuehn** [cirKITers/masKIT#26]
Some optimisers such as the ADAM optimiser have a memory to store recent results as part of their optimisation concept to improve on other methods such as Gradient Descent. However, currently our project does change the number of parameters relevant for optimisation during the training process. This in turn either creates false assumptions with optimisers having a memory as arrays don't map on each other or in the extreme case lead to an exception during training (e.g. out of bounds errors as length of arrays might mismatch).

To cope with this, we might either use `reset` functionality on the optimisers. Based on the assumption that our count of parameters only changes whenever the gradients are close to `0`, we might assume that interim results of optimisers are meaningless. However, this must be checked with the different implementations of optimisers.

Further, we might introduce the capability to *freeze* values to `0` when we initially wanted to remove them, so that the changing action from a given optimiser does not take place and is plainly ignored from our side. However, for this use case it might happen, that interim values represent another reality as the optimiser might assume that its requested change took place. Here we again have to check implementation of optimisers.
